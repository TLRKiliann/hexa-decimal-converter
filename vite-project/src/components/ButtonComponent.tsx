type ButtonProps = {
  convertResult: () => void;
}

function ButtonComponent(props: ButtonProps) {
  return (
    <>
        <button data-testid="testid-btn" type="button" onClick={props.convertResult}>
          Convert
        </button>
    </>
  )
}

export default ButtonComponent