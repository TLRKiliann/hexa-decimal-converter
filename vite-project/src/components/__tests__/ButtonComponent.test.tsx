import { test, expect, vi } from "vitest"
import { create } from "react-test-renderer"
import ButtonComponent from '../ButtonComponent';

//test ok
test('Snapshat of ButtonComponent', () => {
    const tree = create(<ButtonComponent convertResult={function (): void {
        throw new Error("Function not implemented.");
    } } />)
    expect(tree.toJSON()).toMatchSnapshot()
});

test('convertResult return boolean', () => {
    const beverage = "yeah";
    const convertResult = vi.fn(beverage => beverage)
    convertResult(beverage)
    expect(convertResult).toHaveReturnedWith("yeah")
    expect(convertResult).toHaveBeenCalledTimes(1)
});
    
test("testing if convertResult function is present", () => {
    const convertResult = vi.fn();
    expect(convertResult).toBeDefined()
});