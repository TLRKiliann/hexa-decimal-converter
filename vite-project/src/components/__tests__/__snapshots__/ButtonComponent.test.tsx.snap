// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of ButtonComponent 1`] = `
<button
  data-testid="testid-btn"
  onClick={[Function]}
  type="button"
>
  Convert
</button>
`;
